sycl/doc/GettingStartedXilinxFPGA.md

```diff
@@ -528,3 +527,205 @@ the insane! Hopefully...
+ -DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_HOME/sycl \
+ -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_HOME/llvm-spirv \
+ -DLLVM_EXTERNAL_XPTI_SOURCE_DIR=$SYCL_HOME/xpti \
+ -DLLVM_EXTERNAL_LIBDEVICE_SOURCE_DIR=$SYCL_HOME/libdevice \
+ -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl;opencl-aot;xpti;libdevice;libclc" \
+ -DLIBCLC_TARGETS_TO_BUILD="nvptx64--;nvptx64--nvidiacl" \
+ -DSYCL_BUILD_PI_CUDA=ON \
+ -DLLVM_BUILD_TOOLS=ON \
+ -DSYCL_INCLUDE_TESTS=ON \
+ -DLLVM_ENABLE_DOXYGEN=OFF \
+ -DLLVM_ENABLE_SPHINX=OFF \
+ -DBUILD_SHARED_LIBS=ON \
+ -DSYCL_ENABLE_XPTI_TRACING=ON \
+ $SYCL_HOME/llvm
+```
+then build with
+```bash
+cd $SYCL_HOME
+ninja -C build-Release sycl-toolchain
+```
+
+A possible Debug configuration and build script targeting Xilinx FPGA, CUDA & OpenCL:
+```bash
+cd $SYCL_HOME
+mkdir -p "build-Debug" && cd "build-Debug" && cmake \
+ -DCMAKE_BUILD_TYPE="Debug" \
+ -DCMAKE_CXX_FLAGS_DEBUG="-g -fstandalone-debug" \
+ -DLLVM_TABLEGEN="$(pwd)/../build-Release/bin/llvm-tblgen" \
+ -DCLANG_TABLEGEN="$(pwd)/../build-Release/bin/clang-tblgen" \
+ -DCMAKE_INSTALL_PREFIX="../Install-Debug" \
+ -DCMAKE_EXPORT_COMPILE_COMMANDS=1 \
+ -G Ninja \
+ -DLLVM_ENABLE_ASSERTIONS=ON \
+ -DCMAKE_C_COMPILER="/usr/bin/clang-11" \
+ -DCMAKE_CXX_COMPILER="/usr/bin/clang++-11" \
+ -DLLVM_USE_LINKER="lld-11" \
+ -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" \
+ -DLLVM_EXTERNAL_PROJECTS="sycl;llvm-spirv;opencl-aot;xpti;libdevice" \
+ -DLLVM_EXTERNAL_SYCL_SOURCE_DIR=$SYCL_HOME/sycl \
+ -DLLVM_EXTERNAL_LLVM_SPIRV_SOURCE_DIR=$SYCL_HOME/llvm-spirv \
+ -DLLVM_EXTERNAL_XPTI_SOURCE_DIR=$SYCL_HOME/xpti \
+ -DLLVM_EXTERNAL_LIBDEVICE_SOURCE_DIR=$SYCL_HOME/libdevice \
+ -DLLVM_ENABLE_PROJECTS="clang;llvm-spirv;sycl;opencl-aot;xpti;libdevice;libclc" \
+ -DLIBCLC_TARGETS_TO_BUILD="nvptx64--;nvptx64--nvidiacl" \
+ -DSYCL_BUILD_PI_CUDA=ON \
+ -DLLVM_BUILD_TOOLS=ON \
+ -DSYCL_INCLUDE_TESTS=ON \
+ -DLLVM_ENABLE_DOXYGEN=OFF \
+ -DLLVM_ENABLE_SPHINX=OFF \
+ -DBUILD_SHARED_LIBS=ON \
+ -DSYCL_ENABLE_XPTI_TRACING=ON \
+ $SYCL_HOME/llvm
+```
+then build with
+```bash
+cd $SYCL_HOME
+ninja -C build-Debug sycl-toolchain
+```
+
+* `-CMAKE_CXX_FLAGS_RELEASE="-O3 -march=native -fno-omit-frame-pointer"` to have an optimized build with relatively accurate stack traces;
+* `-G Ninja` use ninja instead of make to speedup builds;
+* `-DLLVM_USE_LINKER="lld-11"` select `lld` or `gold` instead of `ld` to speedup builds
+* `-DLLVM_TARGETS_TO_BUILD="X86;NVPTX"` to only build needed targets NVPTX is only needed for CUDA support;
+* `-DLIBCLC_TARGETS_TO_BUILD="nvptx64--;nvptx64--nvidiacl"` needed for CUDA support;
+* `-DSYCL_ENABLE_XPTI_TRACING=ON` adds useful debugging capabilities;
+* `-DBUILD_SHARED_LIBS=ON` to build speedup;
+* `-DLLVM_TABLEGEN="$SYCL_HOME/build-Release/bin/llvm-tblgen" -DCLANG_TABLEGEN="$SYCL_HOME/build-Release/bin/clang-tblgen"` to reuse the tablegen built with the release version and speedup builds. The release build must be done before configuring the Debug one;
```

Comment:
```suggestion
* `-DLLVM_TABLEGEN="$SYCL_HOME/build-Release/bin/llvm-tblgen" -DCLANG_TABLEGEN="$SYCL_HOME/build-Release/bin/clang-tblgen"` to reuse the tablegen output built with the Release version and speedup builds. The Release build must be done before configuring the Debug one;
```

Reply:
I still think that Release and Debug in that case need a capital letter since it refers to the LLVM model.